pennylane/templates/tensornetworks/mps.py

```diff
@@ -0,0 +1,82 @@
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+import re
+
+
+def compute_indices_MPS(wires, loc):
+    """
+    Generate a list of wire indices that quantum gates acts on
+    Args:
+        loc (int): local wire number of a single quantum gate
+        wires (Iterable): the wires on which MPS acts
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+    
+    layers = np.array([[wires[idx] for idx in range(j, j+loc)] for j in range(0, len(wires) - loc // 2, loc // 2)])
+    return layers
+
+
+class MPS_from_function(Operation):
+
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(
+        self,
+        wires,
+        loc,
+        block,
+        n_params_block,
+        weights=None,
+        do_queue=True,
+        id=None,
+    ):
+        n_wires = len(wires)
+        assert loc >= 2, f"loc must be larger than or equal to 2; got {loc}"
+        assert n_wires >= 3, f"number of wires must be greater than or equal to 3; got {n_wires}"
+        assert loc <= n_wires, f"loc must be smaller than or equal to the number of wires; got loc = {loc} and number of wires = {n_wires}"
+  
+        self.n_blocks = int(n_wires / (loc / 2) - 1)
+        self.block = block
+
+        if weights is None:
+            self.weights = np.random.rand(n_params_block, int(self.n_blocks))
+        else:
+            self.weights = weights
+
+        self.ind_gates = compute_indices_MPS(wires, loc)
+
+        # TO DO: raise error if params_block does not match with the input block
+        super().__init__(weights, wires=wires, do_queue=do_queue, id=id)
+
+    def expand(self):
+
+        with qml.tape.QuantumTape() as tape:
+            for idx, w in enumerate(self.ind_gates):
+                self.block(weights=self.weights[idx], wires=w.tolist())
```

Comment:
The way that I see it, the main thing we need for these tensor network templates is a way of generating the wire indices that determine the way in which the blocks are broadcast in the circuit, namely generating the correct `ind_gates`. 

This should make it easy to generalize to other tensor networks.

Reply:
There is a function "compute_indices_MPS" that generates such indices. We have a similar one for the tree tensor network. Do you mean this or something else? (or maybe your comment is already outdated / resolved).